spark_log_parser/loaders/s3.py

```diff
@@ -0,0 +1,90 @@
+import abc
+import boto3
+
+from pathlib import Path
+from urllib.parse import ParseResult, urlparse
+
+from botocore.response import StreamingBody
+
+from spark_log_parser.loaders import AbstractFileDataLoader, BlobFileReaderMixin, LinesFileReaderMixin, \
+    FileChunkStreamWrapper
+
+
+class S3StreamingBodyFileWrapper(FileChunkStreamWrapper):
+    """
+    Small wrapper around botocore.StreamingBody that exposes a lines iterator, in keeping with "pythonic" conventions.
+    botocore.StreamingBody by default iterates over raw chunks of the input stream vs. in a line-delimited manner. This
+    makes some sense, in general, since large files may not have newlines to split on. But as we expect to largely be
+    dealing with line-delimited eventlog files, this default makes more sense for us to use.
+    """
+
+    def __init__(self, body: StreamingBody, **kwargs):
+        super().__init__(**kwargs, chunks=body.iter_chunks())
+
+
+class AbstractS3FileDataLoader(AbstractFileDataLoader, abc.ABC):
```

Comment:
We still do since `read_file_stream` is left unimplemented here. Otherwise, Python thinks this is a concrete class and will complain about unimplemented abstract methods.

Reply:
![Screenshot 2022-12-21 at 1 13 49 PM](https://user-images.githubusercontent.com/117235708/209003767-2b6e02c3-f44c-4c4e-96a9-933643e976d4.png)

Reply:
Could it be a PyCharm thing? VSCode doesn't complain and it behaves as expected:
```
>>> from spark_log_parser.loaders import s3
>>> help(s3.AbstractS3FileDataLoader)
Help on class AbstractS3FileDataLoader in module spark_log_parser.loaders.s3:

class AbstractS3FileDataLoader(spark_log_parser.loaders.AbstractFileDataLoader)
 |  Abstract class that supports loading files directly from S3
...
>>> s3.AbstractS3FileDataLoader()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Can't instantiate abstract class AbstractS3FileDataLoader with abstract method read_file_stream
```
If it helps PyCharm then we can leave it in.

Reply:
aren't you getting an error, though?

> TypeError: Can't instantiate abstract class AbstractS3FileDataLoader with abstract method read_file_stream

^^ this is the exception I'm referring to about Python thinking that this class is concrete, and therefore will require a `read_file_stream` implementation to be available

Reply:
I only get that error when trying to instantiate it, as expected.

Reply:
Ohh, yes, I see what you're saying. And the error message appears to be the same both with and without the `abc.ABC`, so indeed, seems to be unnecessary!